### PR TITLE
tests(smoke): reenable oopif smokes with ToT

### PR DESCRIPTION
**Pull request by adamraine**
@caseq says he landed a change that resolves our attach+waitForDebugger issues. My only verification so far is trying this PR ~3 times in CI without seeing a failure. I will run CI a few more times to verify.

In theory this should:
fix https://github.com/GoogleChrome/lighthouse/issues/13861
fix https://github.com/GoogleChrome/lighthouse/issues/14099
fix https://github.com/GoogleChrome/lighthouse/issues/14093

Knocking on wood because I've been tricked by this before. Either way, any PR that resolves our oopif issues will probably look like this one, so it doesn't hurt to have it ready to go.

### CODE REVIEW

File: .github/workflows/package-test.yml [hunk 1]
```diff
@@ -45,9 +45,14 @@ jobs:
     - run: yarn build-report
     - run: sudo apt-get install xvfb
 
-    # Disabled to fix CI while we investigate what the bug is.
-    #   perf-diagnostics-third-party, oopif-requests, oopif-scripts
-    - run: xvfb-run --auto-servernum bash $GITHUB_WORKSPACE/lighthouse-core/scripts/release/package-test.sh --fraggle-rock --invert-match perf-diagnostics-third-party oopif-requests oopif-scripts
+    - name: Define ToT chrome path
+      run: echo "CHROME_PATH=/home/runner/chrome-linux-tot/chrome" >> $GITHUB_ENV
+
+    - name: Install Chrome ToT
```
Comment (brendankenny) [inline .github/workflows/package-test.yml:51]: should this (and the other cases) include a TODO to eventually move back to stable?

Also, what's left testing stable?

**Review by brendankenny — COMMENTED** [review]
GETTING MY HOPES UP

File: .github/workflows/package-test.yml (hunk 1 above)
Comment (adamraine) [reply .github/workflows/package-test.yml:51]: - Bundled Lighthouse (Legacy navigation)
- Package test (Legacy navigation)
- Chrome stable smokes from the standard matrix

Comment (adamraine) [reply .github/workflows/package-test.yml:51]: > should this (and the other cases) include a TODO to eventually move back to stable?

TBH I'm fine shifting everything (except our explicitly Chrome stable smokes) to use ToT

**Comment by adamraine** [comment]
Just a DevTools / e2e failure, nobody panic

**Comment by adamraine** [comment]
Welp
https://github.com/GoogleChrome/lighthouse/runs/7032720874?check_suite_focus=true

This was the lone failure on `oopif-requests` in ~5 runs, so I'm pretty sure the patch has still *significantly reduced* the likelihood of an oopif request failure but not eliminated it entirely.

`metrics-delayed-fcp` looks like "normal flake" BS unrelated to our current oopif stuff

**Comment by brendankenny** [comment]
> Welp
https://github.com/GoogleChrome/lighthouse/runs/7032720874?check_suite_focus=true

Trying to piece together oopif-requests output in this log makes my brain hurt :)

**Comment by connorjclark** [comment]
> Welp [GoogleChrome/lighthouse/runs/7032720874?check_suite_focus=true](https://github.com/GoogleChrome/lighthouse/runs/7032720874?check_suite_focus=true)
> 
> This was the lone failure on `oopif-requests` in ~5 runs, so I'm pretty sure the patch has still _significantly reduced_ the likelihood of an oopif request failure but not eliminated it entirely.
> 
> `metrics-delayed-fcp` looks like "normal flake" BS unrelated to our current oopif stuff

FYI you can link to an exact line. https://github.com/GoogleChrome/lighthouse/runs/7032720874?check_suite_focus=true#step:12:203

It says a YT .js is missing, but... I do see other YT resources. so, is the expectation just wrong?

> 5 runs
> significantly reduced

FWIW 5 samples is not significant.

File: lighthouse-core/gather/driver.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ class Driver {
    * @private
    * Used to save network and lifecycle protocol traffic. Just Page and Network are needed.
    */
-  _devtoolsLog = new DevtoolsMessageLog(/^(Page|Network)\./);
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply lighthouse-core/gather/driver.js:44]: This was just for debugging. Can you open a new PR?

**Comment by connorjclark** [comment]
So no failures for oopif-scripts?

**Comment by adamraine** [comment]
> It says a YT .js is missing, but... I do see other YT resources. so, is the expectation just wrong?

Yeah I'm not sure. The missing YT .js isn't referenced anywhere in the failing DT log or trace, so maybe the YT embed simply didn't fetch it for some reason?

> Or was it failing 100% of the time before? I can't keep track any more. in that case maybe it is significant :P

The `perf-diagnostics-third-party` was failing 99% of the time or thereabouts. The new `oopif-request` was failing pretty often too. If you take a look at the CI failures on [`d74b655` (#14153)](https://github.com/GoogleChrome/lighthouse/pull/14153/commits/d74b655750f12d88346f87044d6044bc281ec3d5) you can see just how common those failures were *before* switching to ToT Chrome.

FWIW each "run" contains multiple points of failure (i.e. Smoke FR, Smoke Bundled FR, package FR, DT smoke * 3 flaky smokes * 5 runs). So there were ~50-60 problematic smoke tests that could have resulted in a failure but only 1 actually had a failure.

File: lighthouse-core/gather/driver.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulirish) [reply lighthouse-core/gather/driver.js:44]: yeah im pretty sure i messed up a conflict resolution merge.  thx for the driveby

**Comment by adamraine** [comment]
TODO:
- Everything will be ToT except Package-test and stable-specific CLI smokes
- Version check on oopif network requests
- Circle back and decide where it's best to test stable/ToT

File: lighthouse-core/scripts/download-chrome.sh [hunk 3]
```diff
@@ -32,5 +32,5 @@ fi
 if [ -e "$CHROME_PATH" ]; then
   echo "cached chrome found"
 else
-  wget "$url" --no-check-certificate -q -O chrome.zip && unzip -q chrome.zip
+  curl "$url" -Lo chrome.zip && unzip -q chrome.zip
```
Comment (adamraine) [inline lighthouse-core/scripts/download-chrome.sh:35]: wget is not in windows but curl is

**Review by brendankenny — APPROVED** [review]
we really need #14197, #14135, and eliminating legacy mode to make the union of our test matrices simpler to follow, but LGTM! Happy to have these back so soon

**Pull request merged**